tests/unit/utils/sparqlhandler/test_data_cube_inspector.py

```diff
@@ -902,3 +906,73 @@ def test_get_columns_for_component_attribute_pivoted():
     expected_titles = ["Imports Status", "Exports Status"]
 
     assert actual_titles == expected_titles
+
+
+def test_get_primary_csv_url():
+    """
+    Testing that the csv_url for the primary CSV defined in the data cube CSV-W is correctly retrieved.
+    """
+    csvw_metadata_json_path = (
+        _test_case_base_dir
+        / "single-unit_single-measure"
+        / "energy-trends-uk-total-energy.csv-metadata.json"
+    )
+    data_cube_inspector = get_data_cube_inspector(csvw_metadata_json_path)
+    csv_url = data_cube_inspector.get_primary_csv_url()
+
+    assert csv_url == "energy-trends-uk-total-energy.csv"
+
+
+def test_load_pandas_df_from_standard_shape_csv_url():
+    """
+    Testing that a dataframe with columns represented in the correct data types
+    can be loaded from a standard shape CSVW and that a distinction is made between
+    attribute reasource and attribute literal columns.
+    """
+    csvw_metadata_json_path = (
+        _test_case_base_dir
+        / "inspector-load-dataframe"
+        / "standard-shape"
+        / "standard-shape-out"
+        / "testing-converting-a-standard-shape-csvw-to-pandas-dataframe.csv-metadata.json"
+    )
+    data_cube_inspector = get_data_cube_inspector(csvw_metadata_json_path)
+    csv_url = data_cube_inspector.get_primary_csv_url()
+
+    dataframe, validation_errors = data_cube_inspector.get_dataframe(csv_url)
+
+    assert isinstance(dataframe, pd.DataFrame)
+    assert dataframe["Dim1"].dtype == "string"
+    assert dataframe["Dim2"].dtype == "string"
+    assert dataframe["Dim1"].dtype == "string"
+    assert dataframe["Attr Resource"].dtype == "string"
+    assert dataframe["Attr Literal"].dtype == "Int64"
+    assert dataframe["Units"].dtype == "string"
+    assert dataframe["Measures"].dtype == "string"
+    assert dataframe["Obs"].dtype == "short"
+    assert not any(validation_errors)
+
+
+def test_load_pandas_df_from_pivoted_shape_csv_url():
+    """
+    Testing that a dataframe with columns represented in the correct data types
+    can be loaded from a pivoted shape CSVW.
+    """
+    csvw_metadata_json_path = (
+        _test_case_base_dir
+        / "inspector-load-dataframe"
+        / "pivoted-shape"
+        / "pivoted-shape-out"
+        / "testing-converting-a-pivoted-csvw-to-pandas-dataframe.csv-metadata.json"
+    )
+    data_cube_inspector = get_data_cube_inspector(csvw_metadata_json_path)
+    csv_url = data_cube_inspector.get_primary_csv_url()
+
+    dataframe, validation_errors = data_cube_inspector.get_dataframe(csv_url)
+
+    assert isinstance(dataframe, pd.DataFrame)
+    assert dataframe["Dim1"].dtype == "string"
+    assert dataframe["Obs1"].dtype == "string"
```

Comment:
Is this the one that has a obs val column which has the time data type?

Reply:
Yeah it was the `time` column. Have added a comment to explain why we expect "string"